subsys/bluetooth/audio/ascs.c

```diff
@@ -671,10 +671,13 @@ static void ascs_iso_recv(struct bt_iso_chan *chan,
 		return;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_BAP_DEBUG_STREAM_DATA) &&
-	    ep->status.state != BT_BAP_EP_STATE_STREAMING) {
-		LOG_DBG("ep %p is not in the streaming state: %s", ep,
-			bt_bap_ep_state_str(ep->status.state));
+	if (ep->status.state != BT_BAP_EP_STATE_STREAMING) {
+		if (IS_ENABLED(CONFIG_BT_BAP_DEBUG_STREAM_DATA)) {
+			LOG_DBG("ep %p is not in the streaming state: %s", ep,
+				bt_bap_ep_state_str(ep->status.state));
+		}
+
+		net_buf_unref(buf);
```

Comment:
Is the `net_buf_unref` correct to use here? IIRC it is no, and if it is, it needs to be added to _all_ cases.

Reply:
That's surprise :) I see that there is no unref in samples as well. I thought that stack passes the buffer ownership

Reply:
The stack unrefs the buf in `bt_iso_recv` by calling `bt_conn_reset_rx_state(iso);`